internal/fs/tarfs/tarfs.go

```diff
@@ -135,8 +136,16 @@ func (tfs *TarFS) indexEntries() error {
 		header, err := tr.Next()
 		if err != nil {
 			if errors.Is(err, io.EOF) {
+				if len(tfs.entries) == 0 {
+					// indicates the given file is empty
+					return fmt.Errorf("%s: file is empty: %w", tfs.path, errdef.ErrInvalidTarFile)
+				}
 				break
 			}
+			if errors.Is(err, io.ErrUnexpectedEOF) {
+				// indicates that its either not a tarfile or it is a corrupted one
+				return fmt.Errorf("%s: %w", tfs.path, errdef.ErrInvalidTarFile)
+			}
```

Comment:
The `io.ErrUnexpectedEOF` error itself indicates the tar file is corrupted. There is no need to wrap it to `errdef.ErrInvalidTarFile`.